### ISSUE TITLE
Launch Template support

### ISSUE THREAD

**Issue by symonds** [AutoSpotting/AutoSpotting#285]
When running autospotting against an ASG which is configured using Launch Templates instead of Launch Configurations the current master 24a5693f0c336f32fddce4fdaab87aeccca26d4f branch has the following error:

```
2018/08/01 12:36:28 instance.go:331: Launching spot instance with a bid = 0.1
2018/08/01 12:36:28 instance.go:312: Couldn't launch spot instance: InvalidParameterCombination: A launch template ID and a launch template name cannot be specified in the same request.
status code: 400, request id: 786d65a8-b6c1-4489-ad73-42ee66ef268f
2018/08/01 12:36:28 autoscaling.go:337: Could not launch cheapest spot instance: InvalidParameterCombination: A launch template ID and a launch template name cannot be specified in the same request.
status code: 400, request id: 786d65a8-b6c1-4489-ad73-42ee66ef268f
2018/08/01 12:36:28 instance.go:312: Couldn't launch spot instance: InvalidParameterCombination: A launch template ID and a launch template name cannot be specified in the same request.
status code: 400, request id: dd109c9c-4a26-4535-ae1d-b1377a7b3ce7
2018/08/01 12:36:28 autoscaling.go:337: Could not launch cheapest spot instance: InvalidParameterCombination: A launch template ID and a launch template name cannot be specified in the same request.
status code: 400, request id: dd109c9c-4a26-4535-ae1d-b1377a7b3ce7
2018/08/01 12:36:28 instance.go:312: Couldn't launch spot instance: InvalidParameterCombination: A launch template ID and a launch template name cannot be specified in the same request.
status code: 400, request id: b3ab266e-839c-4f05-aa48-a858d8693ae6
2018/08/01 12:36:28 autoscaling.go:337: Could not launch cheapest spot instance: InvalidParameterCombination: A launch template ID and a launch template name cannot be specified in the same request.
status code: 400, request id: b3ab266e-839c-4f05-aa48-a858d8693ae6
2018/08/01 12:36:28 Execution completed, nothing left to do
END RequestId: 81d0857d-9587-11e8-af34-2b4da098a743
```

Seems to be related to https://github.com/cristim/autospotting/blob/master/core/instance.go#L422

**Comment by cristim** [comment]
Good catch, thanks for testing this. 

I wasn't aware that just one of those is required and I have no environments based on launch templates where I could test this.

Try to delete the line with the name, and please create a PR if it works as expected.

**Comment by cristim** [comment]
@symonds have a look at this fork by @snake-aas, apparently he's also working on this: https://github.com/snake-aas/autospotting

**Comment by xbglowx** [comment]
I got something working, but need to fix some tests. Once that is done, I will create a PR: https://github.com/xbglowx/autospotting/commit/570193b4d4450228e8231deeee19981b0293d7d2

**Comment by cristim** [comment]
Thanks @xbglowx!

I'm looking forward for your PR. Let me know if you need any help, I'm most of the time reachable on gitter.im

**Comment by bi0morph** [comment]
Hey guys!
Are there any updates about supporting the `Launch Template`?

**Comment by cristim** [comment]
I would love to see this feature land but the PR we had for it still needs some work. If you need this feature feel free to pick it up and help get it through. Thanks!

**Comment by cristim** [comment]
I just updated #303 which should properly add support for launch templates, but didn't test it yet since I don't have such a setup.

Can anyone test this?

**Comment by cristim** [comment]
Closing after someone confirmed the code change in #303 fixes this issue.

**Comment by rezmuh** [comment]
@cristim I'm having issue with Launch template as well. If I installed autospotting just by installing evaluation binaries (through https://bit.ly/AutoSpotting), does it include #303 ? Thanks

**Comment by cristim** [comment]
@rezmuh yes, the current evaluation binaries should work, but please let me know if you notice any issues.

**Comment by rezmuh** [comment]
@cristim i just reinstall autospotting using evaluation binaries but it doesn't seem to work, still.

Previously, I could see the logs in CloudWatch and would be able to see the same errors as the OP stated. However, after I re-installed it, I didn't see any logs created in cloudwatch.

Also, I am using opt-out if that helps.

**Comment by cristim** [comment]
Thanks, but to me that looks like an unrelated bug.

Please report it on another issue with all the required information and I may look into it within the next weeks.

**Comment by cristim** [comment]
@rezmuh, your issue should be addressed in #352, please test that and let me know if you still notice any failures

**Comment by rezmuh** [comment]
@cristim how do I test that? it doesn't look like it's in the nightly build yet, is it?

**Comment by cristim** [comment]
@rezmuh try again in a few minutes
